compiler/env/OMRClassEnv.hpp

```diff
@@ -83,6 +83,20 @@ class OMR_EXTENSIBLE ClassEnv
    bool isPrimitiveClass(TR::Compilation *comp, TR_OpaqueClassBlock *clazz) { return false; }
    bool isAnonymousClass(TR::Compilation *comp, TR_OpaqueClassBlock *clazz) { return false; }
    bool isValueTypeClass(TR_OpaqueClassBlock *) { return false; }
+
+   /**
+    * \brief
+    *    Checks whether the specified class has fields that are value types that have not been expanded
+    *    (flattened) into their own components.
+    *
+    * \param clazz
+    *    The class that is to be checked for unflattened value type fields
+    *
+    * \return
+    *    `true` if the specified class has unexpanded value type fields;
+    *    `false` otherwise
+    */
+   bool hasUnflattenedValueTypeField(TR_OpaqueClassBlock *clazz) { return false; }
```

Comment:
This is the mechanics of the check, but is this the semantics? I imagine, knowing what I know of OpenJ9, that what you mean is the type trivially initializable or is the object initializable with all zero values. Would it be better for the query to capture that semantics rather than the mechanics which seems to be what this is saying?

Reply:
Good point.  How about something like `isZeroInitializable` or `areInstancesZeroInitializable`?

Reply:
I think isZeroInitializable probably makes the most sense

Reply:
Revised in commit bc9a120d4c02e5d269a7d5476f5b5c2290d3d930